pkg/cloudmeta/metadata_test.go

```diff
@@ -0,0 +1,142 @@
+// Copyright (C) 2017 ScyllaDB
+
+package cloudmeta
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"testing"
+)
+
+func TestGetInstanceMetadata(t *testing.T) {
+	t.Run("when there is no active providers", func(t *testing.T) {
+		cloudmeta := &CloudMeta{}
+
+		meta, err := cloudmeta.GetInstanceMetadata(context.Background())
+		if !errors.Is(err, ErrNoProviders) {
+			t.Fatalf("expected err, got: %v", err)
+		}
+
+		if meta.InstanceType != "" {
+			t.Fatalf("meta.InstanceType should be empty, got %v", meta.InstanceType)
+		}
+
+		if meta.CloudProvider != "" {
+			t.Fatalf("meta.CloudProvider should be empty, got %v", meta.CloudProvider)
+		}
+	})
+
+	t.Run("when there is only one active provider", func(t *testing.T) {
+		cloudmeta := &CloudMeta{
+			providers: []CloudMetadataProvider{newTestProvider(t, "test_provider_1", "x-test-1", nil)},
+		}
+
+		meta, err := cloudmeta.GetInstanceMetadata(context.Background())
+		if err != nil {
+			t.Fatalf("unexpected err: %v", err)
+		}
+
+		if meta.InstanceType != "x-test-1" {
+			t.Fatalf("meta.InstanceType should be 'x-test-1', got %v", meta.InstanceType)
+		}
+
+		if meta.CloudProvider != "test_provider_1" {
+			t.Fatalf("meta.CloudProvider should be 'test_provider_1', got %v", meta.CloudProvider)
+		}
+	})
+
+	t.Run("when there is more than one active provider", func(t *testing.T) {
+		cloudmeta := &CloudMeta{
+			providers: []CloudMetadataProvider{
+				newTestProvider(t, "test_provider_1", "x-test-1", nil),
+				newTestProvider(t, "test_provider_2", "x-test-2", nil),
+			},
+		}
+
+		// Only first one should be returned.
+		meta, err := cloudmeta.GetInstanceMetadata(context.Background())
+		if err != nil {
+			t.Fatalf("unexpected err: %v", err)
+		}
+
+		if meta.InstanceType != "x-test-1" {
+			t.Fatalf("meta.InstanceType should be 'x-test-1', got %v", meta.InstanceType)
+		}
+
+		if meta.CloudProvider != "test_provider_1" {
+			t.Fatalf("meta.CloudProvider should be 'test_provider_1', got %v", meta.CloudProvider)
+		}
+	})
+	t.Run("when there is more than one active provider, but first returns err", func(t *testing.T) {
+		cloudmeta := &CloudMeta{
+			providers: []CloudMetadataProvider{
+				newTestProvider(t, "test_provider_1", "x-test-1", fmt.Errorf("'test_provider_1' err")),
+				newTestProvider(t, "test_provider_2", "x-test-2", nil),
+			},
+		}
+
+		// Only first succesfull one should be returned.
+		meta, err := cloudmeta.GetInstanceMetadata(context.Background())
+		if err != nil {
+			t.Fatalf("unexpected err: %v", err)
+		}
+
+		if meta.InstanceType != "x-test-2" {
+			t.Fatalf("meta.InstanceType should be 'x-test-2', got %v", meta.InstanceType)
+		}
+
+		if meta.CloudProvider != "test_provider_2" {
+			t.Fatalf("meta.CloudProvider should be 'test_provider_2', got %v", meta.CloudProvider)
+		}
+	})
```

Comment:
There is quite much of repetitions in this test code.
I suggest to follow Table-Driven Tests, where you just define what's given and what's expected with the struct.
Example:
```
    tests := []struct {
        name           string
        providers      []CloudMetadataProvider
        expectedError  error
        expectedMeta   InstanceMetadata
    }{
        {
            name:           "no active providers",
            providers:      nil,
            expectedError:  ErrNoProviders,
            expectedMeta:   InstanceMetadata{},
        },
            for _, tt := range tests {
        t.Run(tt.name, func(t *testing.T) {
            cloudmeta := &CloudMeta{providers: tt.providers}
            meta, err := cloudmeta.GetInstanceMetadata(context.Background())

            if !errors.Is(err, tt.expectedError) {
                t.Fatalf("expected error %v, got %v", tt.expectedError, err)
            }

            if meta != tt.expectedMeta {
                t.Fatalf("expected metadata %+v, got %+v", tt.expectedMeta, meta)
            }
        })
    }
    ```

Reply:
yes, it's just my preference to avoid table driven tests as they usually lack of readability. 
I'll update this to align with the rest of the project :)